doc/help/elektra-glossary.md

```diff
@@ -102,6 +102,34 @@
 - [Metadata](elektra-metadata.md):
   Allows us to describe configuration settings.
 
+- **Namespace Root Key**:
+  A namespace root key, is the root key of a namespace.
+  It consists only of a namespace, but no other parts.
+  e.g.: `system:/`, `user:/`, `/`
+
+- **Child** (of key K):
+  A key whose name starts the same as that of K, but has exactly one extra part.
+
+- **Parent** (of key K):
+  A key P where K is a child of P.
+  A namespace root key has no parent, every other key has exactly one parent.
+
+- **Descendant** (of key K):
+  A key whose name starts the same as that of K, with some extra parts.
+  In other words: A child of K, or a child of a child of K, etc.
+
+- **Ancestor** (of key K):
+  A key X where K is a descendant of X.
+  In other words: The parent of K, or the parent of the parent of K, etc.
+
+- **Key Hierarchy**:
+  A key hierarchy is a collection of keys, which are all descendants of a single key R.
+  This key R is called the _root key_ of the hierarchy.
```

Comment:
Unclear if a key hierarchy is all keys of a namespace or everything descendant of a specific key k. If it is the second, we don't need the term, as we already have descendant? (Btw. collection also would need to be defined).

If it is the first:

```suggestion
- **Key Hierarchy**:
  A key hierarchy are all keys, which are all descendants of the *namespace root key*.
```

Reply:
> collection also would need to be defined

I'm not going to start a dictionary. I'm a software engineer, not a linguist.

I did not use "collection" with any specific definition in mind (e.g. `java.util.Collection`). It's just a group/set/assembly/... of things.

> Unclear if a key hierarchy is all keys of a namespace or everything descendant of a specific key k.

The definition says very clearly what a "key hierarchy" is.

All keys in a namespace would be an example of a key hierarchy, where the root key is the namespace root.